vm/devices/storage/disk_nvme/nvme_driver/src/namespace.rs

```diff
@@ -499,6 +504,52 @@ impl Namespace {
 
         Ok(())
     }
+
+    /// Return Namespace ID.
+    pub fn nsid(&self) -> u32 {
+        self.nsid
+    }
+
+    /// Save namespace data for servicing.
+    pub fn save(&self) -> anyhow::Result<SavedNamespaceData> {
+        let id = self.state.identify.lock();
+        let mut save_data = SavedNamespaceData {
+            nsid: self.nsid,
+            block_count: self.state.block_count.load(Ordering::Relaxed),
+            identify_ns: [0; 4096],
+        };
+        save_data.identify_ns.copy_from_slice(id.as_bytes());
+        Ok(save_data)
+    }
+
+    /// Restore namespace data after servicing.
+    pub(super) fn restore(
```

Comment:
I can't really find where this is used. We do restore a namespace, but it seems like that namespace object never gets used. Did I miss something?

Reply:
When later we call get_namespace() which calls get_driver(), and get_driver() will find an existing entry in self.devices.entry(pci_id)

This restore adds the object to the list here:
self.devices.insert(disk.pci_id.clone(), nvme_driver);

Reply:
And then `get_namespace()` will call `driver.namespace(...)`, which will create a new `Namespace`, overwriting this one.

Reply:
Let me check if I lost some code while porting, or just double check the logic.

Reply:
Looks like a bug, there is an overwrite of the restored namespace object. Fixing.